isis/src/base/objs/PvlKeyword/PvlKeyword.cpp

```diff
@@ -1391,8 +1393,24 @@ namespace Isis {
           KEYWORD
 
         Well, no value/units may exist so we're done processing this keyword.
+        Now check if it's a reserved keyword that is valueless
        */
-      return 1; // Valid & Successful
+      bool allowedValueless = false;
+      std::vector<QString> reservedKeywordNames = {"OBJECT",
+                                                   "BEGIN_OBJECT",
+                                                   "END_OBJECT",
+                                                   "ENDOBJECT",
+                                                   "GROUP",
+                                                   "END_GROUP",
+                                                   "ENDGROUP",
+                                                   "END"};
+      QString keywordNameUpper = keywordName.toUpper();
+      for (auto reservedKeywordName : reservedKeywordNames) {
+        if (keywordNameUpper == reservedKeywordName) {
```

Comment:
Consider `auto const &reservedKeywordName` and swap the order of the variables in if test for const safety.

Reply:
I'm not entirely sure what reversing the comparison gains? How is there increased const safety when you switch them given the other changes requested?

Reply:
It's just a best practice thing. I try to ensure equality tests in if statements always have the const variable/constant (if there is one) on the left side in case the `==` winds up as `=`. It's then an assignment statement that often compiles and executes without any indication of a potential bug. Some compilers may issue a warning for it nowdays.

Reply:
I took a bit of a middle ground approach. I left the if logic the same but made `keywordNameUpper` const so the potential `=` instead of the `==` is caught.